### PR TITLE
first draft for new GS

### PR DESCRIPTION
**Pull request by jrobinAV**
Here is a proposal for the new getting started content. 

- I tried to be short and direct in both the content and the style, but it can undoubtedly be improved.
- We can also improve the style, creating a better graphic for the DAG, better screenshots, etc...

Please review the English, the form, the content, and the style

image
image
image
image
image

### CODE REVIEW

File: docs/getting_started-new/hello_world.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import taipy as tp
+from taipy import Config, Core, Gui
+
+
+################################################################
+#                  Configure your application                  #
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world.py]: I would have said "THE" application

File: docs/getting_started-new/hello_world.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+name_data_node_cfg = Config.configure_data_node(id="input_name")
+# A second data node configuration represents the message to print
+message_data_node_cfg = Config.configure_data_node(id="message")
+# The task represents the build_message function
+build_msg_task_cfg = Config.configure_task("build_msg", build_message, name_data_node_cfg, message_data_node_cfg)
+# The scenario represent the whole execution graph
+scenario_cfg = Config.configure_scenario("scenario", task_configs=[build_msg_task_cfg])
+
+################################################################
+#                     Design your interface                    #
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world.py]: THE user interface

File: docs/getting_started-new/hello_world.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Name: <|{input_name}|input|>
+
+<|submit|button|on_action=submit_scenario|>
+
+Message: <|{message}|text|>
+"""
+
+if __name__ == "__main__":
+    ################################################################
+    #            Instantiate the Core service and run it           #
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world.py]: => Instantiate and run the Core service

File: docs/getting_started-new/hello_world.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    ################################################################
+    Core().run()
+
+    ################################################################
+    #               Manage your scenarios and data nodes           #
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world.py]: I see the idea in manage.
What about: Create the data flow?

File: docs/getting_started-new/hello_world.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ################################################################
+    hello_scenario = tp.create_scenario(scenario_cfg)
+
+    ################################################################
+    #            Instantiate the Gui service and run it            #
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world.py]: Instantiate and run the Gui service

File: docs/getting_started-new/hello_world_scenario.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import taipy as tp
```
Comment (FabienLelaquais) [inline docs/getting_started-new/hello_world_scenario.py:1]: Same comment as in hello_world.py

File: docs/getting_started-new/index.md [hunk 7]
```diff
@@ -1,7 +1,180 @@
...
+To get started with Taipy, ensure that you have Python (version 3.8 or higher) and
+[pip](https://pip.pypa.io) installed on your system. Once you have these prerequisites, 
+open your terminal or command prompt and run the following command to download and
+install the latest stable release of Taipy.
+
+``` console
+$ pip install taipy
+```
+
+!!! info "Other Installation Options"
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: => "Other installation options"

File: docs/getting_started-new/index.md [hunk 8]
```diff
@@ -1,7 +1,180 @@
...
+
+name_data_node_cfg = Config.configure_data_node(id="input_name")
+message_data_node_cfg = Config.configure_data_node(id="message")
+build_msg_task_cfg = Config.configure_task("build_msg", build_message, name_data_node_cfg, message_data_node_cfg)
+scenario_cfg = Config.configure_scenario("scenario", task_configs=[build_msg_task_cfg])
+```
+
+- Lines 4-5 define the function that the task will use during execution.
+- Lines 8-9 configure the two data nodes, *input_name* and *message*.
+- Line 10 configures a task named *build_msg* associated with the `build_message()`
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: => \*build_message()\*

File: docs/getting_started-new/index.md [hunk 9]
```diff
@@ -1,7 +1,180 @@
...
+  function, specifying input and output data nodes.
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: "specifying input" -> "specifying the input"

File: docs/getting_started-new/index.md [hunk 10]
```diff
@@ -1,7 +1,180 @@
...
+- Finally, line 11 configures the execution graph of the scenario providing 
+  the previously configured task.
+
+### Running The Core service
+
+Running the Core service allows Taipy to process the configuration made in the
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: The Core service "processes the configuration made in the  previous step"?

File: docs/getting_started-new/index.md [hunk 11]
```diff
@@ -1,7 +1,180 @@
...
+
+if __name__ == "__main__":
+    Core().run()
+```
+Line 3 is standard boilerplate code to ensure that your script runs only from the 
+main module, preventing accidental execution. It is strongly recommended to include it.
+
+In line 4, we simply instantiate and run a Core service.
+
+### Creating Scenarios and accessing data
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: No capital to Scenarios

File: docs/getting_started-new/index.md [hunk 12]
```diff
@@ -1,7 +1,180 @@
...
+
+Now that Taipy Core service is running, you can create and manage scenarios, submit
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: "that Taipy Core service" -> "that the Taipy Core service"

File: docs/getting_started-new/index.md [hunk 13]
```diff
@@ -1,7 +1,180 @@
...
+
+In line 4, we simply instantiate and run a Core service.
+
+### Creating Scenarios and accessing data
+
+Now that Taipy Core service is running, you can create and manage scenarios, submit 
+task graphs for execution, and access data nodes. Here's a code snippet to demonstrate 
+these actions:
+
+
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: You may want to remove those two empty lines.
The code snippet is attached to the paragraph before.

File: docs/getting_started-new/index.md [hunk 14]
```diff
@@ -1,7 +1,180 @@
...
+  the previously configured task.
+
+### Running The Core service
+
+Running the Core service allows Taipy to process the configuration made in the 
+previous step and set up the scenario management feature.
+
+Include the following code in your Python script:
+
+``` python linenums="1"
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: The convention you use basically everywhere is not to have this white space after the three backticks

File: docs/getting_started-new/index.md [hunk 15]
```diff
@@ -1,7 +1,180 @@
...
+import taipy as tp
+
+hello_scenario = tp.create_scenario(scenario_cfg)
+hello_scenario.input_name.write("Taipy")
+hello_scenario.submit()
+print(hello_scenario.message.read())
+```
+
+In line 3, method `tp.create_scenario()` instantiates the new scenario name 
+`hello_scenario` from the scenario configuration built before.
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: \*hello_scenario\*

File: docs/getting_started-new/index.md [hunk 16]
```diff
@@ -1,7 +1,180 @@
...
+
+Line 4, sets the input data node *input_name* of `hello_scenario` with the string
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: same

File: docs/getting_started-new/index.md [hunk 17]
```diff
@@ -1,7 +1,180 @@
...
+hello_scenario.input_name.write("Taipy")
+hello_scenario.submit()
+print(hello_scenario.message.read())
+```
+
+In line 3, method `tp.create_scenario()` instantiates the new scenario name 
+`hello_scenario` from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of `hello_scenario` with the string 
+value "Taipy" using the write() method.
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: \`write()\`

File: docs/getting_started-new/index.md [hunk 18]
```diff
@@ -1,7 +1,180 @@
...
+
+Line 5 submits the `hello_scenario` for execution, which triggers the creation
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: same

File: docs/getting_started-new/index.md [hunk 19]
```diff
@@ -1,7 +1,180 @@
...
+
+In line 3, method `tp.create_scenario()` instantiates the new scenario name 
+`hello_scenario` from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of `hello_scenario` with the string 
+value "Taipy" using the write() method.
+
+Line 5 submits the `hello_scenario` for execution, which triggers the creation 
+and execution of a job. This job reads the input data node, passes the value 
+to the `build_message()` function, and writes the result to the output data node.
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: same
... then in the rest of the text body.

Only API elements should have a fixed size font.

File: docs/getting_started-new/index.md [hunk 20]
```diff
@@ -1,7 +1,180 @@
...
+
+Line 6 reads and prints the output data node `message` that has been written by 
+the execution of the scenario `hello_scenario`.
+
+Putting all together the pieces, here is the Python code corresponding to the example:
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: "here is the Python code corresponding to the example" -> "you can download the entire Python code for the example from \[this link\]\(hello_world_scenario.py\)"

File: docs/getting_started-new/index.md [hunk 21]
```diff
@@ -1,7 +1,180 @@
...
+<a href="./hello_world_scenario.py" download>`hello_world_scenario.py`</a>
+
+And here is the expected output.
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md]: "output." -> "output:"

File: docs/getting_started-new/index.md [hunk 22]
```diff
@@ -1,7 +1,180 @@
...
+```
+
+TODO: explain the code above and result below
+
+![result](result.png){width=50%}
+
+
+Putting all together the scenario and the Gui pieces, here is the complete python code 
+corresponding to the example:
+<a href="./hello_world.py" download>`hello_world.py`</a>
```
Comment (FabienLelaquais) [inline docs/getting_started-new/index.md:166]: same as above.

**Review by FabienLelaquais — APPROVED** [review]
Minor optional comments

File: docs/getting_started-new/hello_world.py (hunk 1 above)
Comment (erikfTW) [reply docs/getting_started-new/hello_world.py]: Can be even: Configure application.

File: docs/getting_started-new/hello_world.py [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+    return f"Hello {name}!"
+
+
+# A first data node configuration represents a name
+name_data_node_cfg = Config.configure_data_node(id="input_name")
+# A second data node configuration represents the message to print
+message_data_node_cfg = Config.configure_data_node(id="message")
+# The task represents the build_message function
+build_msg_task_cfg = Config.configure_task("build_msg", build_message, name_data_node_cfg, message_data_node_cfg)
+# The scenario represent the whole execution graph
```
Comment (erikfTW) [inline docs/getting_started-new/hello_world.py]:
```suggestion
# The scenario represents the whole execution graph
```

File: docs/getting_started-new/hello_world.py (hunk 3 above)
Comment (erikfTW) [reply docs/getting_started-new/hello_world.py]: or maybe: Create the Core service and start it.

File: docs/getting_started-new/hello_world.py (hunk 4 above)
Comment (erikfTW) [inline docs/getting_started-new/hello_world.py]:
```suggestion
    #               Manage scenarios and data nodes           #
```

File: docs/getting_started-new/hello_world.py (hunk 5 above)
Comment (erikfTW) [reply docs/getting_started-new/hello_world.py]: here as well maybe: Create the GUI service and start it.

File: docs/getting_started-new/hello_world.py [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+################################################################
+
+hello_scenario = None
+input_name = "Taipy"
+message = None
+
+
+def submit_scenario(state):
+    state.hello_scenario.input_name.write(state.input_name)
+    state.hello_scenario.submit(wait=True)
```
Comment (FlorianJacta) [inline docs/getting_started-new/hello_world.py]: `wait=True` is not necessary

**Review by FlorianJacta — APPROVED** [review]

File: docs/getting_started-new/index.md [hunk 25]
```diff
@@ -1,7 +1,180 @@
...
+hide:
+  - navigation
+---
 
-Next, we will delve into the two fundamental components of web application development using Python in Taipy. You will gain insights into managing HTTP requests, routing, database management, and the creation of dynamic web pages through templating engines. Additionally, we will explore topics such as user authentication, data validation, and best practices to fortify the robustness and security of your applications.
+# Getting Started with Taipy
 
-!!! info "For additional information about the Taipy open-source Python library, please refer to [Welcome to Taipy](https://docs.taipy.io/en/develop/)"
+Welcome to Taipy! This guide will walk you through the process of installing Taipy 
+and building your first application.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
Welcome to Taipy! This guide will show you how to install Taipy 
and create your first application.
```

File: docs/getting_started-new/index.md [hunk 26]
```diff
@@ -1,7 +1,180 @@
...
 
-Throughout this article, we will furnish code examples and practical insights to bolster your comprehension and facilitate the practical application of your newfound knowledge. Furthermore, we will offer recommendations for supplementary resources, tutorials, and community support to enrich your learning journey.
+## Install Taipy with pip
+
+To get started with Taipy, ensure that you have Python (version 3.8 or higher) and
+[pip](https://pip.pypa.io) installed on your system. Once you have these prerequisites, 
+open your terminal or command prompt and run the following command to download and
+install the latest stable release of Taipy.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
To begin with Taipy, ensure you have Python installed (version 3.8 or higher) along with
[pip](https://pip.pypa.io) on your system. Once you have these prerequisites, 
open your terminal or command prompt and run the following command to get the latest stable version of Taipy downloaded and installed:
```

File: docs/getting_started-new/index.md [hunk 27]
```diff
@@ -1,7 +1,180 @@
...
+
+``` console
+$ pip install taipy
+```
+
+!!! info "Other Installation Options"
+    
+    If you need alternative installation methods or if you don't have Python 
+    or pip installed, please refer to the [installation page](installation/index.md)
+    for additional instructions.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
    You can find additional instructions on the [installation page](installation/index.md) 
    if you require different ways to install Taipy or don't have Python or pip installed.
```

File: docs/getting_started-new/index.md [hunk 28]
```diff
@@ -1,7 +1,180 @@
...
+
+
+## First Taipy Scenario 
+Taipy introduces the concept of a *Scenario* to model pipeline executions. 
+A *Scenario* consists of an execution graph, often represented as a Directed 
+Acyclic Graph (DAG), where tasks (or functions) exchange data. Your scenario 
+can be as simple or as complex as your needs require.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
Acyclic Graph (DAG), where tasks (or functions) exchange data. You can make your 
scenario as easy or as complicated as you want, depending on what you need.
```

File: docs/getting_started-new/index.md [hunk 29]
```diff
@@ -1,7 +1,180 @@
...
+
+For your first scenario, we'll create a simple "Hello World" example 
+to demonstrate the configuration process. Here's an illustration of the 
+scenario's execution graph made of two data nodes (blue boxes) and one task 
+(orange box):
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
For your first scenario, we'll make a straightforward "Hello World" example 
to show you how the setup works. Here's a picture of the
scenario's execution graph, which has two data nodes (blue boxes) and one task 
(orange box):
```

File: docs/getting_started-new/index.md [hunk 30]
```diff
@@ -1,7 +1,180 @@
...
+
+![hello world example](hello_world.svg){width=50%}
+
+It consists in one input data node named *name*. Then a task named 
+*build_message* takes the first data node and returns a second 
+data node named *message*.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
This involves a single input data node called *name*.  Afterward, a task called 
*build_message* takes the first initial data node and produces a second  
data node called *message*.
```

File: docs/getting_started-new/index.md [hunk 31]
```diff
@@ -1,7 +1,180 @@
...
+    return f"Hello {name}!"
+
+
+name_data_node_cfg = Config.configure_data_node(id="input_name")
+message_data_node_cfg = Config.configure_data_node(id="message")
+build_msg_task_cfg = Config.configure_task("build_msg", build_message, name_data_node_cfg, message_data_node_cfg)
+scenario_cfg = Config.configure_scenario("scenario", task_configs=[build_msg_task_cfg])
+```
+
+- Lines 4-5 define the function that the task will use during execution.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md:64]:
```suggestion
- Lines 4-5 define the task's function during execution.
```

File: docs/getting_started-new/index.md [hunk 32]
```diff
@@ -1,7 +1,180 @@
...
+
+Include the following code in your Python script:
+
+``` python linenums="1"
+from taipy import Core
+
+if __name__ == "__main__":
+    Core().run()
+```
+Line 3 is standard boilerplate code to ensure that your script runs only from the
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
Line 3 is standard boilerplate code to ensure your script runs only from the 
```

File: docs/getting_started-new/index.md [hunk 33]
```diff
@@ -1,7 +1,180 @@
...
+``` python linenums="1"
+from taipy import Core
+
+if __name__ == "__main__":
+    Core().run()
+```
+Line 3 is standard boilerplate code to ensure that your script runs only from the 
+main module, preventing accidental execution. It is strongly recommended to include it.
+
+In line 4, we simply instantiate and run a Core service.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
In line 4, we instantiate and run a Core service.
```

File: docs/getting_started-new/index.md [hunk 34]
```diff
@@ -1,7 +1,180 @@
...
+
+### Scenario creation and data access
+
+Now that Taipy Core service is running, you can create and manage scenarios, submit 
+task graphs for execution, and access data nodes. Here's a code snippet to demonstrate 
+these actions:
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
With the Taipy Core service up and running, you have the ability to create and manage scenarios, submit 
task graphs for execution, and access data nodes. Here's a code snippet to illustrate  
these actions:
```

File: docs/getting_started-new/index.md [hunk 35]
```diff
@@ -1,7 +1,180 @@
...
+`hello_scenario` from the scenario configuration built before.
+
+Line 4, sets the input data node *input_name* of `hello_scenario` with the string 
+value "Taipy" using the write() method.
+
+Line 5 submits the `hello_scenario` for execution, which triggers the creation 
+and execution of a job. This job reads the input data node, passes the value 
+to the `build_message()` function, and writes the result to the output data node.
+
+Line 6 reads and prints the output data node `message` that has been written by
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
Line 6 reads and prints the output data node `message` written by 
```

File: docs/getting_started-new/index.md [hunk 36]
```diff
@@ -1,7 +1,180 @@
...
+``` console
+[2023-02-08 20:19:35,062][Taipy][INFO] job JOB_build_msg_9e5a5c28-6c3e-4b59-831d-fcc8b43f882e is completed.
+Hello Taipy!
+```
+
+## Graphical Interface
+
+In the previous step, you used Taipy Python APIs to access, edit, and submit a scenario. 
+This interaction is commonly implemented within Python functions triggered 
+by end-user actions in a graphical interface built using Taipy.
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
This interaction is typically integrated into Python functions that get triggered
by end-user actions in a graphical interface built using Taipy. 
```

File: docs/getting_started-new/index.md [hunk 37]
```diff
@@ -1,7 +1,180 @@
...
+```
+
+## Graphical Interface
+
+In the previous step, you used Taipy Python APIs to access, edit, and submit a scenario. 
+This interaction is commonly implemented within Python functions triggered 
+by end-user actions in a graphical interface built using Taipy. 
+
+In this section we propose a basic graphical interface to interact with the scenario 
+created before. Use the following Python code :
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
In this section, we suggest a simple graphical interface for interacting with the previously created 
scenario. Use the following Python code:
```

File: docs/getting_started-new/index.md [hunk 38]
```diff
@@ -1,7 +1,180 @@
...
+    Gui(page).run()
+```
+
+TODO: explain the code above and result below
+
+![result](result.png){width=50%}
+
+
+Putting all together the scenario and the Gui pieces, here is the complete python code 
+corresponding to the example:
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
Combining both the scenario and the GUI components, here's the full Python code 
for the example:
```

File: docs/getting_started-new/index.md [hunk 39]
```diff
@@ -1,7 +1,180 @@
...
+<a href="./hello_world.py" download>`hello_world.py`</a>
+
+This concludes the basic steps to get you started with Taipy. For more realistic 
+use cases and advanced features, refer to the
```
Comment (erikfTW) [inline docs/getting_started-new/index.md]:
```suggestion
That covers the fundamental steps to begin with Taipy. For more realistic 
use cases and advanced features, refer to 
```

**Review by erikfTW — CHANGES_REQUESTED** [review]

**Comment by vhfgosselin** [comment]
Ok I will look into it but usually our competitors provide more
explanations about setting it using ends and differentenvs. For instance:
https://docs.streamlit.io/library/get-started/installation

**Pull request merged**